code/game/machinery/computer/security.dm

```diff
@@ -1,46 +1,321 @@
 //This file was auto-corrected by findeclaration.exe on 25.5.2012 20:42:31
 
-/obj/structure/machinery/computer/secure_data//TODO:SANITY
+/obj/structure/machinery/computer/secure_data
 	name = "Security Records"
 	desc = "Used to view and edit personnel's security records"
 	icon_state = "security"
 	req_access = list(ACCESS_MARINE_BRIG)
 	circuit = /obj/item/circuitboard/computer/secure_data
-	var/obj/item/card/id/scan = null
-	var/obj/item/device/clue_scanner/scanner = null
-	var/rank = null
-	var/screen = 1
-	var/datum/data/record/active1 = null
-	var/datum/data/record/active2 = null
-	var/a_id = null
-	var/temp = null
-	var/printing = null
-	var/can_change_id = 0
-	var/list/Perp
-	var/tempname = null
-	//Sorting Variables
-	var/sortBy = "name"
-	var/order = 1 // -1 = Descending - 1 = Ascending
-
-/obj/structure/machinery/computer/secure_data/attackby(obj/item/O as obj, user as mob)
-
-	if(istype(O, /obj/item/device/clue_scanner) && !scanner)
-		var/obj/item/device/clue_scanner/S = O
-		if(!S.print_list)
-			to_chat(user, SPAN_WARNING("There are no prints stored in \the [S]!"))
+	var/obj/item/device/clue_scanner/scanner
+
+	// For optimisation, we keep track of the viewed record and only send detailed data for it.
+	var/active_record_ref
+	var/datum/data/record/active_record_general
+	var/datum/data/record/active_record_security
+
+/obj/structure/machinery/computer/secure_data/attackby(obj/item/held_object as obj, user as mob)
+	if(istype(held_object, /obj/item/device/clue_scanner) && !scanner)
+		var/obj/item/device/clue_scanner/held_scanner = held_object
+		if(!held_scanner.print_list)
+			to_chat(user, SPAN_WARNING("There are no prints stored in \the [held_scanner]!"))
 			return
 
 		if(usr.drop_held_item())
-			O.forceMove(src)
-			scanner = O
-			to_chat(user, "You insert [O].")
+			held_scanner.forceMove(src)
+			scanner = held_scanner
+			to_chat(user, "You insert [held_scanner].")
+			update_static_data_for_all_viewers()
 
 	..()
 
 /obj/structure/machinery/computer/secure_data/attack_remote(mob/user as mob)
 	return attack_hand(user)
 
-//Someone needs to break down the dat += into chunks instead of long ass lines.
+/obj/structure/machinery/computer/secure_data/ui_data(mob/user)
```

Comment:
All return paths for this proc should return something. Rather than explicitly having data as a list variable, just make the default return value an empty list and append data to it.

Reply:
I used to do it that way and then I think Nanu taught me to do it this way 😆
I can change to the      .["blah"] = xyz      way if you prefer